pkg/database/databaseLayer.go

```diff
@@ -48,7 +48,13 @@ func (db DBClient) UpdateUser(user models.User) error {
 
 func (db DBClient) GetUserOrders(user models.User) ([]models.OrderItem, error) {
 	result := []models.OrderItem{}
-	err := db.Db.Model([]models.OrderItem{}).Where("user_id=?", user.ID).Find(&result)
+	err := db.Db.Preload("Orders.OrderItems").First(&user, "id = ?", user.ID)
```

Comment:
Check for ways to optimize this, you don't want to preload orders. You want to query just order items but use a join query to apply the filter on user_id.